src/main/java/org/openrewrite/java/migrate/guava/PreferJavaUtilOptionalOrElseNull.java

```diff
@@ -0,0 +1,89 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.guava;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class PreferJavaUtilOptionalOrElseNull extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Prefer `java.util.Optional#orElse(null)` over `com.google.common.base.Optional#orNull()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces `com.google.common.base.Optional#orNull()` with `java.util.Optional#orElse(null)`.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(5);
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return new HashSet<>(Arrays.asList("RSPEC-4738", "guava"));
+    }
+
+    @Override
+    protected UsesType<ExecutionContext> getApplicableTest() {
+        return new UsesType<>("com.google.common.base.Optional");
+    }
+
+    @Override
+    protected UsesMethod<ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesMethod<>("com.google.common.base.Optional orNull()");
+    }
+
+    @Override
+    protected JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new PreferJavaUtilOptionalOrElseNullVisitor();
+    }
+
+    private static class PreferJavaUtilOptionalOrElseNullVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private static final MethodMatcher guavaCreateTempDirMatcher = new MethodMatcher("com.google.common.base.Optional orNull()");
+
+        @Override
+        public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext executionContext) {
+            J.CompilationUnit c = super.visitCompilationUnit(cu, executionContext);
+            maybeAddImport("java.util.Optional");
+            maybeRemoveImport("com.google.common.base.Optional");
+            return c;
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            J.MethodInvocation mi = super.visitMethodInvocation(method, executionContext);
+            if (guavaCreateTempDirMatcher.matches(mi)) {
+                return mi.withName(mi.getName().withSimpleName("orElse"))
+                        .withTemplate(JavaTemplate.builder(this::getCursor, "null").build(), mi.getCoordinates().replaceArguments());
```

Comment:
I suspect this is actually the problem of the `ChangeType` recipe (I will investigate this further, when I find time), but as you can see in the following screenshot from the debugger, the type attribution of the `optional` parameter refers to the correct new type, but as its owner still refers to the old `JavaType.Method` object: 
<img width="1291" alt="image" src="https://user-images.githubusercontent.com/408765/227859803-22d4752d-eef6-4dbd-b79b-8750ca0af6e0.png">